tools/driver/autolink_extract_main.cpp

```diff
@@ -106,32 +107,44 @@ class AutolinkExtractInvocation {
   }
 };
 
-// Look inside the binary 'Bin' and append any linker flags found in its
-// ".swift1_autolink_entries" section to 'LinkerFlags'. If 'Bin' is an archive,
-// recursively look inside all children within the archive. Return 'true' if
-// there was an error, and 'false' otherwise.
+/// Look inside the object file 'ObjectFile' and append any linker flags found in
+/// its ".swift1_autolink_entries" section to 'LinkerFlags'.
+static void
+extractLinkerFlagsFromObjectFile(const llvm::object::ObjectFile *ObjectFile,
+                                 std::vector<std::string> &LinkerFlags) {
+  // Search for the section we hold autolink entries in
+  for (auto &Section : ObjectFile->sections()) {
+    llvm::StringRef SectionName;
+    Section.getName(SectionName);
+    if (SectionName == ".swift1_autolink_entries") {
```

Comment:
I quoted the description of the name field in `3. Section Table (Section Headers)` from [MS PECOFF](https://msdn.microsoft.com/en-us/windows/hardware/gg463119.aspx).
```
... For longer names, this field contains a slash (/) that is followed by an ASCII 
representation of a decimal number that is an offset into the string table.
Executable images do not use a string table and do not support section names longer
than 8 characters. Long names in object files are truncated if they are emitted to 
an executable file.
```
Because the section '.swift1_autolink_entries' will not be emitted to an executable file, we can use the long name for autolink feature. But we can not use the long names '.swift2_protocol_conformances' and  '.swift2_type_metadata' instead of '.sw2prtc' or '.sw2tymd', because they are emitted to an executable and loaded into the system memory.